### ISSUE TITLE
TKO learning path feedback

### ISSUE THREAD

**Issue by jp-gouin** [vmware-tanzu-learning/tko-ref-arch-training#53]
**Let us know what feedback you have**

**Full disclosure : I took the learning path one month ago so the below feedback might not be up to date**

### Startup : 
I’m  a little bit confuse , the course assume that I have the CKA level and we go to the introduction to containers
Same the description of the learning path and the readme does not really match 

### Container : 
Honestly it’s an interesting course.
Personally (but that my only though on this :) ) , I won’t talk to much about docker but rather container or OCI , it will make more sense as Kubernetes has deprecated the docker engine from their supported list of engines
-> Maybe mention containerd -> actual engine of Kubernetes
-> Podman to experiment (deamonless)
-> Skopeo for build integrated in CI for instance (or buildah)
 
What is actually awesome here is the experimentation right after the course, and I’m not sure many training site offer this quality level 

### Kubernetes Fundamentals : 
-> That could be a course to prepare for the CKA or CKD ?

 
## TKO Products and Components 

TKO shows (through our learning path or yours) that it’s a set of solutions that can be used and deployed in multiple ways.
Could there be an official way , documentation, to use TKO with all its solutions? 
-> Use TMC to deploy TKG clusters and do this to enable the integration for TO and TSM… 
-> This is how you connect TO and TSM and so on 
So that we address TKO as a whole and not isolated pieces ?

### TKG
There is a focus on TKGm rather than TKGs. But i don’t think it really matter in TKO architecture

### TMC
A little bit quick in the overview of all features 
There are some attention point that we see during the build of the learning path (Integration configuration, Backup using restic etc)
No mention of the CLI to automate actions on TMC

### TO
I think a good intro would be to resume what is observability , why it”s needed in today architecture
It’s 15min but the video is 3min ?

### TSM:
Quick start :) 
The intro of TSM is actually a really good intro for Observability as well
In my opinion on major key of TSM is the capability to split an application across multiple cluster in multiple region and not have to worry about the interconnection between services

### Antrea : 
This is very interesting stuff and I actually learned several things :) 
But is it really related to TKO ? Maybe for the Prometheus part but I think that should go deeper and have a dashboard in TO using Prometheus metrics

### Carvel: 
To be honest I didn’t get through the 1h30 about carvel tooling :) but I will !
What’s the outcome here ? And what’s the relation with TKO 
From a learner point of view the duration of Carvel video is 1h30 and TO is 30min…
-> For TAP it would make sense to deep dive into carvel packages as Cartographer mainly use this solution with Kapp

### TMC
Great description of security and policies
Backup and restore : 
-> Enable data protection ? What does that do ? (Deploy Velero using restic) 
-> Target location -> need to be S3 compliant (AWS, Minio, Ceph, …) 
-> you don’t need to have a external facing S3 endpoint as velero is deployed locally 
     -> This is worth mentioning as customer care about where datas are located 
     -> The backup belong to the customer and is stored in his infrastructure

### TSM : 
There is no need to redevelop the app for TSM but the deployment need Istio CRD which is not standard 
-> How to deploy TSM ? There is some gotchas here about the size of the cluster to be able to host TSM 
You need to edit directly Istio CRD to manage the traffic (for canary or A/B testing for instance) , is it supported by TSM ?

### TO :
 Speaking of correlation , one thing that does not TO is the correlation between trace/monitoring data and log …
-> How to deploy TO in a TKO arch ?
-> How to send a trace to TO ? -> The proxy offers a OpenTelemetry endpoint that can be used by customer applications to send metrics
-> Alerting ?
-> Define SLO, SLI ?
-> Link between TSM and TO out of the box ?

## Test Drive
No shortcut on the desktop
No kubectl cli installed 

## General
Videos are great , how fast it is to shoot and add a new video 
- For a new feature ?
- For a deep dive in a feature ?
I think that our textual learning path and the TKO KubeAcademy are complementary
We can have on one side quick and small iteration to update a textual learning path 
-> We can measure if a user/customer is reading some specific parts of the learning path and create a video about it and enrich the TKO (or other) learning path on KubeAcademy

**Comment by djschny** [comment]
Thanks so much for taking the time to share this with us. It's a mix of things where some have been addressed, some are things not specific to TKO, some are new issues identified etc. Many of the things are material that we have not developed but were created by others and we are pointing too.

> I’m a little bit confuse , the course assume that I have the CKA level and we go to the introduction to containers

We were never supposed to include that entry level items but were asked to after the fact. We shouldn't have obliged. Details on this can be found discussed [here](https://github.com/vmware-tanzu-learning/tko-ref-arch-training/issues/38)

> Same the description of the learning path and the readme does not really match

We are a victim of the reference guide changing the name. The learning path name was made first and then the README was made and the name was updated in the docs, we will make it consistent and update, thanks!

> Personally (but that my only though on this :) ) , I won’t talk to much about docker but rather container or OCI , it will make more sense as Kubernetes has deprecated the docker engine from their supported list of engines

This is a pre-existing course and we have it [tracked](https://github.com/vmware-tanzu-learning/edu-ilt-foundations/issues/206) but haven't acted yet. We can re-record and adjust that and other references, but it hasn't been a priority compared to working on later on TKO items.

> -> Maybe mention containerd -> actual engine of Kubernetes
> -> Podman to experiment (deamonless)
> -> Skopeo for build integrated in CI for instance (or buildah)

We try to not bring in other tools in the course to keep things as simple as possible when a person is trying to grasp concepts.

> What is actually awesome here is the experimentation right after the course, and I’m not sure many training site offer this quality level

Thanks!

> Kubernetes Fundamentals :
> -> That could be a course to prepare for the CKA or CKD ?

Are you referring to [Kubernetes Core Concepts](https://kube.academy/paths/kubernetes-core-concepts) learning path?

> Could there be an official way , documentation, to use TKO with all its solutions?
Yes there is we are working to get closer to that. The main ask of the team initially was to create training to allow folks to be able to successfully deploy the TKO reference architecture, so that's what has been focused on so far.


> So that we address TKO as a whole and not isolated pieces ?
This is intentional. Later level material and other items will address that and are being created by others. Our goal is to intentionally help people with all the isolated pieces separately and incrementally so that way when they actually go to deploy and use TKO, it's not overwhelming having to learn everything at once.

> There is a focus on TKGm rather than TKGs. But i don’t think it really matter in TKO architecture
Correct, we are trying avoid the TKG[a-z] as much as possible

For the `TKO Products and Components` in general, those were links to pre-existing material made by others that we put in to have something. It's not ideal in many situations but none were made by this team so unfortunately was out of our control until we replace them with more suitable content.

> It’s 15min but the video is 3min ?

The duration that shows in KA interface unfortunately is a manual field that needs to be set. It's probably leftover from a previous video we had and then replaced with the current one. Will fix that, thank you!

> Antrea : This is very interesting stuff and I actually learned several things :) But is it really related to TKO ?
We were provided feedback by learners that they had trouble with Antrea when implementing the deployment guides so we added it.

> To be honest I didn’t get through the 1h30 about carvel tooling What’s the outcome here ?

This [has been fixed](https://github.com/vmware-tanzu-learning/tko-ref-arch-training/issues/41) by providing better clarification to links that are optional/supplemental material versus "required" material.

> And what’s the relation with TKO
The deployment guides and components of TKO unfortunately use primarily `ytt` and `kapp` and assume the person knows about these tools. The intent/relation was to help make sure people know about the tools and are comfortable using them prior to deploying TKO.

> Test Drive
That lab was written by a different team and we just linked to it. I will see if there is a way to get the feedback to that team.

> how fast it is to shoot and add a new video
Pretty quick. The majority of the work is all the work prior.
